### ISSUE TITLE
Incorrect IDE0044 (Make field readonly) when used with "ref"

### ISSUE THREAD

**Issue by zlatanov** [dotnet/roslyn#38995]
**Version Used**: 16.3.1

**Steps to Reproduce**:
``` csharp
using System;

class Program
{
    private static Int32 Value;

    static void Main()
    {
        ref var value = ref Value;
        value += 1;
    }
}
```
**Expected Behavior**: No errors / warnings / messages.

**Actual Behavior**:  IDE0044	Make field readonly.

Here is a counter example where it works as expected:
``` csharp
using System;

class Program
{
    private static Int32 Value;

    static void Main()
    {
        Increment( ref Value );
    }

    static void Increment( ref Int32 value ) => value += 1;
}
```

**Comment by AlFasGD** [comment]
Reproducible in VS 16.11.1
